SharedCode/src/commonMain/kotlin/com/plusmobileapps/sharedcode/redux/AppState.kt

```diff
@@ -0,0 +1,54 @@
+package com.plusmobileapps.sharedcode.redux
+
+import com.plusmobileapps.sharedcode.FeedRepository
+import com.plusmobileapps.sharedcode.RedditPostResponse
+import com.plusmobileapps.sharedcode.di.commonModule
+import org.kodein.di.direct
+import org.kodein.di.instance
+import org.reduxkotlin.*
+
+data class AppState(
+    val isLoading: Boolean = false,
+    val posts: List<RedditPostResponse> = listOf(),
+    val error: String? = null
+) {
+    companion object {
+        val initialState = AppState(isLoading = true)
+    }
+}
+
+class Actions {
+    object LoadingFeed
+    data class FeedLoaded(val posts: List<RedditPostResponse>)
+    data class FeedError(val message: String)
+}
+
+object ApplicationStarted
+object FeedStarted
+data class UpVoteAction(val postId: String)
+data class DownVoteAction(val postId: String)
+
+fun rootReducer(state: AppState, action: Any): AppState {
+    return when (action) {
+        is ActionTypes.INIT -> state
+        is ApplicationStarted -> state
+        is FeedStarted -> {
```

Comment:
Clean up some of these actions as they were just tests to see if the store was in fact working.